File: backend/app/schemas/assistant.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class AssistantMessageRequest(BaseModel):
    text: str = Field(min_length=1)
    channel: str = "internal"


class AssistantActionOut(BaseModel):
    id: int
    action_type: str
    status: str
    payload_json: dict[str, Any]
    result_json: dict[str, Any]
    created_at: datetime
    confirmed_at: datetime | None = None

    class Config:
        from_attributes = True


class AssistantMessageResponse(BaseModel):
    conversation_id: int
    reply: str
    action: AssistantActionOut | None = None


class TelegramBindRequest(BaseModel):
    chat_id: str = Field(min_length=1)
    username: str | None = None


class AssistantActionResult(BaseModel):
    id: int
    status: str
    result_json: dict[str, Any]

File: frontend/src/api/assistant.ts
import api from './client';

export interface AssistantAction {
  id: number;
  action_type: string;
  status: string;
  payload_json: Record<string, any>;
  result_json: Record<string, any>;
  created_at: string;
  confirmed_at?: string | null;
}

export interface AssistantMessageResponse {
  conversation_id: number;
  reply: string;
  action?: AssistantAction | null;
}

export interface AssistantActionResult {
  id: number;
  status: string;
  result_json: Record<string, any>;
}

export const sendAssistantMessage = async (text: string, channel = 'internal') => {
  const { data } = await api.post<AssistantMessageResponse>('/assistant/messages', { text, channel });
  return data;
};

export const listAssistantActions = async () => {
  const { data } = await api.get<AssistantAction[]>('/assistant/actions');
  return data;
};

export const confirmAssistantAction = async (id: number) => {
  const { data } = await api.post<AssistantActionResult>(`/assistant/actions/${id}/confirm`);
  return data;
};

export const cancelAssistantAction = async (id: number) => {
  const { data } = await api.post<AssistantActionResult>(`/assistant/actions/${id}/cancel`);
  return data;
};

export const bindCurrentTelegramChat = async (chatId: string, username?: string) => {
  const { data } = await api.post<{ status: string; chat_id: string }>('/assistant/telegram/bind-current', {
    chat_id: chatId,
    username: username || undefined,
  });
  return data;
};
